src/providers/filters/entry.ts

```diff
@@ -3,7 +3,7 @@ import { Entry, EntryFilterFunction, MicromatchOptions, Pattern, PatternRe } fro
 import * as utils from '../../utils';
 
 export default class EntryFilter {
-	public readonly index: Map<string, undefined> = new Map();
+	public readonly index: Map<unknown, undefined> = new Map();
```

Comment:
I'm also curious if we could have a shared/cached index in case we want to create multiple separate fast-glob instances (such as would be needed to support ordered glob negation)